engine/crates/federated-graph/src/federated_graph/v4.rs

```diff
@@ -122,6 +134,73 @@ pub enum Value {
     List(Box<[Value]>),
 }
 
+#[derive(serde::Serialize, serde::Deserialize, Clone)]
+pub struct Object {
```

Comment:
I had to redefine these in v4.rs because they contain `Key`s, which contain the new type. That cascaded.